020-Getting-started/090-remix.mdx

```diff
@@ -0,0 +1,797 @@
+    post
+  });
+}
+```
+
+Use the values returned from the `loader` within the `V2_MetaFunction` function to display the Post's title
+within the page's title. This function differs from the landing page version by taking a parameter containing a `data` property.
+The value in `data` is the payload returned from the `loader` function. So, the Post's `title` value can be extracted and used:
+
+```tsx
+export const meta: V2_MetaFunction<typeof loader> = ({ data }) => {
+  return [{ title: `${data?.post?.title} - Xata and Remix` }];
+};
+```
+
+Finally, the main page function makes use of the `userLoaderData` hook to retrieve the single Post information and uses the values,
+`title`, `pubDate`, and `description`, for the Post are added to the UI:
+
+```tsx {2,11,13,15}
+export default function Post() {
+  const { post } = useLoaderData<typeof loader>();
+
+  return (
+    <div className="w-full max-w-5xl mt-16">
+      <p className="mb-2">
+        <a href="/" className="text-purple-600">
+          &larr; Back to blog
+        </a>
+      </p>
+      <h1 className="text-3xl mb-2">{post?.title}</h1>
+      <p className="text-sm mb-4 text-purple-950 dark:text-purple-200">
+        {new Date(post?.pubDate ?? '').toDateString()}
+      </p>
+      <p className="text-xl">{post?.description}</p>
+    </div>
+  );
+}
+```
+
+The single blog post page will look as follows:
+
+![Single blog post](images/getting-started-blog-post.png)
+
+## Search posts
+
+The last piece of functionality to add to the application is full-text fuzzy search of blog posts.
+
+When you insert data into a Xata database, it is automatically indexed for full-text search. So you don't need to change
+any configuration to enable search, just need to use the
+[TypeScript SDK search feature](https://xata.io/docs/typescript-client/search).
+
+Let's add this functionality to the landing page:
+
+```tsx title="app/routes/_index.tsx" {1,7-24,27,31-40}
+import { LoaderArgs, V2_MetaFunction, json } from '@remix-run/node'
+import { useLoaderData } from '@remix-run/react'
+import { getXataClient } from 'src/xata'
+
+...
+
+export async function loader({ request }: LoaderArgs) {
+  const xata = getXataClient()
+
+  const url = new URL(request.url)
+  const search = url.searchParams.get('q')
+
+  let posts = null
+  if (search) {
+    posts = await xata.db.Posts.search(search, { fuzziness: 2 })
+  } else {
+    posts = await xata.db.Posts.getAll()
+  }
+
+  return json({
+    posts,
+    search,
+  })
+}
+
+export default function Index() {
+  const { posts, search } = useLoaderData<typeof loader>()
+
+  return (
+    <>
+      <div className="w-full max-w-5xl mt-16">
+        <form>
+          <input
+            name="q"
+            defaultValue={search || ''}
+            placeholder="Search..."
+            className="w-full rounded-lg p-2 border-2 dark:text-purple-950"
+          />
+        </form>
+      </div>
+
+      ...
+    </>
+  )
+}
+```
+
+Add search with these changes.
```

Comment:
I liked it better when you intro-ed this with "Let's break down what's happening in the code above." in your other guides. The way this reads, it seems like there is more code for me to add below, rather than you explaining bits of the code above. I think this is unique to this guide?

Reply:
Updated 👍